src/SprykerEco/Zed/CrefoPay/Persistence/CrefoPayRepository.php

```diff
@@ -0,0 +1,212 @@
+use Generated\Shared\Transfer\PaymentCrefoPayTransfer;
+use Generated\Shared\Transfer\SpySalesOrderItemEntityTransfer;
+use Orm\Zed\CrefoPay\Persistence\SpyPaymentCrefoPayOrderItemQuery;
+use Orm\Zed\CrefoPay\Persistence\SpyPaymentCrefoPayQuery;
+use Orm\Zed\Sales\Persistence\SpySalesOrderItemQuery;
+use Spryker\Zed\Kernel\Persistence\AbstractRepository;
+use Spryker\Zed\PropelOrm\Business\Runtime\ActiveQuery\Criteria;
+
+/**
+ * @method \SprykerEco\Zed\CrefoPay\Persistence\CrefoPayPersistenceFactory getFactory()
+ */
+class CrefoPayRepository extends AbstractRepository implements CrefoPayRepositoryInterface
+{
+    /**
+     * @param string $reference
+     *
+     * @return \Generated\Shared\Transfer\PaymentCrefoPayTransfer
+     */
+    public function getPaymentCrefoPayByReference(string $reference): PaymentCrefoPayTransfer
+    {
+        $query = $this->getPaymentCrefoPayQuery()->filterByReference($reference);
+        $entityTransfer = $this->buildQueryFromCriteria($query)->findOne();
+
+        if ($entityTransfer === null) {
+            return new PaymentCrefoPayTransfer();
+        }
+
+        return $this->getFactory()
+            ->createCrefoPayPersistenceMapper()
+            ->mapEntityTransferToPaymentCrefoPayTransfer($entityTransfer, new PaymentCrefoPayTransfer());
+    }
+
+    /**
+     * @param int $idSalesOrder
+     *
+     * @return \Generated\Shared\Transfer\PaymentCrefoPayTransfer
+     */
+    public function getPaymentCrefoPayByIdSalesOrder(int $idSalesOrder): PaymentCrefoPayTransfer
+    {
+        $query = $this->getPaymentCrefoPayQuery()->filterByFkSalesOrder($idSalesOrder);
+        $entityTransfer = $this->buildQueryFromCriteria($query)->findOne();
+
+        if ($entityTransfer === null) {
+            return new PaymentCrefoPayTransfer();
+        }
+
+        return $this->getFactory()
+            ->createCrefoPayPersistenceMapper()
+            ->mapEntityTransferToPaymentCrefoPayTransfer($entityTransfer, new PaymentCrefoPayTransfer());
+    }
+
+    /**
+     * @param string $pspReference
+     *
+     * @return \Generated\Shared\Transfer\PaymentCrefoPayTransfer
+     */
+    public function getPaymentCrefoPayByPspReference(string $pspReference): PaymentCrefoPayTransfer
+    {
+        $query = $this->getPaymentCrefoPayQuery()->filterByPspReference($pspReference);
+        $entityTransfer = $this->buildQueryFromCriteria($query)->findOne();
+
+        if ($entityTransfer === null) {
+            return new PaymentCrefoPayTransfer();
+        }
+
+        return $this->getFactory()
+            ->createCrefoPayPersistenceMapper()
+            ->mapEntityTransferToPaymentCrefoPayTransfer($entityTransfer, new PaymentCrefoPayTransfer());
+    }
+
+    /**
+     * @param int $idSalesOrder
+     *
+     * @return \Generated\Shared\Transfer\PaymentCrefoPayOrderItemTransfer[]
+     */
+    public function getAllPaymentCrefoPayOrderItemsByIdSalesOrder(int $idSalesOrder): array
```

Comment:
according to agreement in core, this (and other similar methods) should be called `findXXxx`